ros_buildfarm/templates/release/rpm/mock_config.cfg.em

```diff
@@ -0,0 +1,50 @@
+include('/etc/mock/default.cfg')
+
+# Change the root name since we're modifying the chroot
+config_opts['root'] += '-ros-buildfarm'
+
+# Add python3-rpm-macros to resolve %{python3_pkgversion} in rosdep rules
+config_opts['chroot_setup_cmd'] += ' python3-rpm-macros'
+
+# Install weak dependencies to get group members
+config_opts['yum_builddep_opts'] = config_opts.get('yum_builddep_opts', []) + ['--setopt=install_weak_deps=True']
+config_opts['dnf_builddep_opts'] = config_opts.get('dnf_builddep_opts', []) + ['--setopt=install_weak_deps=True']
+config_opts['microdnf_builddep_opts'] = config_opts.get('microdnf_builddep_opts', []) + ['--setopt=install_weak_deps=True']
+
+# Disable debug packages until infrastructure can handle it
+config_opts['macros']['debug_package'] = '%{nil}'
+
+# Hack the %{dist} macro to allow release suffixing
+config_opts['macros']['%dist'] = '.' + config_opts['dist'] + '%{?dist_suffix}'
+
+# Required for running mock in Docker
+config_opts['use_nspawn'] = False
+
+@[if os_name in ['centos', 'rhel'] and os_code_name == '7']@
+# Inject g++ 8 into RHEL 7 builds
+config_opts['chroot_setup_cmd'] += ' devtoolset-8-gcc-c++ devtoolset-8-make-nonblocking'
+config_opts['macros']['_buildshell'] = '/usr/bin/scl enable devtoolset-8 -- /bin/sh'
```

Comment:
This section is noteworthy. I'm injecting the devtoolset-8 SCL for RHEL 7 to get a new enough version of GCC. Doing it this way means that we don't need special spec file logic in Bloom for RHEL 7.

Additionally, I'm expecting a special-built version of `make` that has a patch taken from upstream to make one of the operations non-blocking. I discovered this issue in the ci.ros2.org jobs, and the workaround package seems to keep the calls from consistently hanging.

That package is expected to be in the bootstrap repository. It provides a virtual package with the same name as the original package, so it is interchangeable with the one upstream, but the upstream one is still installed by default. It would be nice to get RedHat to address the issue, but I'm not holding my breath.